polytope/solvers.py

```diff
@@ -0,0 +1,119 @@
+"""Interface to linear programming solvers.
+
+By default, for linear programming the `polytope` package selects
+the fastest solver that it finds installed. You can change this
+default by setting the variable `lp_solver` in the module
+`_solvers`. For example:
+
+```python
+from polytope import _solvers
+
+_solvers.lp_solver = 'scipy'
+```
+
+Choose an installed solver to avoid errors.
+"""
+# Copyright (c) 2011-2017 by California Institute of Technology
```

Comment:
I recommend moving this copyright declaration to the top of the file, which will keep it consistent with other source files in this package. Doing so will not affect docstring parsing; e.g., the output of `pydoc polytope.solvers` is the same.

Reply:
I am fine with moving the copyright notice to the top, but before applying this change, I should note the reasons for selecting this location:

1. Avoiding accidents where copy/pasting from a file accidentally omits the first copyright line, which contains the year and copyright holder's name. I considered this scenario after encountering an instance of it (which was along other files that retained the full notice--so it must have been an accident).

2. Placing the module docstring as close to the file's top as possible, so that reading with `head solvers.py` will present the docstring first, and less scrolling will be needed to view the docstring in an editor. Of course, this concern is less relevant for 2-3 lines of copyright notice. It is relevant when the entire license (or a notice of substantial size, as e.g., in GPL'ed software) is included in a file.

Also, I do not know whether the location of a copyright notice within a file has legal implications. In particular, whether only lines below the copyright notice are regarded as subject to the notice. I think that this is not the case, since copyright applies in principle even in absence of a notice. In any case, only the module's docstring would be left out.

Lately, I favor "3-clause BSD" more, as it avoids the ambiguity of what "-3" means (is it a version specifier? is it a subtraction of 3 things? maybe of all the 3 clauses?). (An example similar to this wording is https://github.com/tulip-control/gr1c/blob/b373f93fe1e66499590e4868db1ba5ec226737ee/README.md).

Reply:
You provide fair arguments for keeping the notice where it is. I do not have a strong preference either way, but being consistent in the package is important. Note that there are no legal consequences for being at the top of the file, nor is it necessary to include copyright and license declarations in every file. I am also happy if we skip this question for this PR, and revisit it for the entire `polytope` package afterward.

The Software Freedom Law Center provides a good treatment of these concerns at <http://softwarefreedom.org/resources/2012/ManagingCopyrightInformation.html>. In particular, read the section "Comparing two systems: file-scope and centralized notices".

Reply:
a9986781e5f5f29df5ee1bce8d822cda6c7285e5 moves the notice to the file's top, to ensure consistency among files. We can revisit this question on its own later.

Thanks for linking to the SFLC's advice, I read that page, and there are also other useful resources in that website that I want to study.